### PR TITLE
enh: add tests for checkListenerPorts

### PR DESCRIPTION
**Pull request by sakshi-1505**
**What this PR does / why we need it**:
It increase unit test coverage for LB file

**Which issue this PR fixes(if applicable)**:
refers https://github.com/kubernetes/cloud-provider-openstack/issues/2400

**Special notes for reviewers**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/openstack/loadbalancer_test.go [hunk 1]
```diff
@@ -534,3 +534,158 @@ func Test_getListenerProtocol(t *testing.T) {
 		})
 	}
 }
+
+func TestLbaasV2_checkListenerPorts(t *testing.T) {
+	type fields struct {
+		LoadBalancer LoadBalancer
+	}
```
Comment (dulek) [inline pkg/openstack/loadbalancer_test.go:541]: Doesn't seem you ever use this? Can we remove it?

File: pkg/openstack/loadbalancer_test.go [hunk 2]
```diff
@@ -534,3 +534,158 @@ func Test_getListenerProtocol(t *testing.T) {
...
+	type args struct {
+		service            *corev1.Service
+		curListenerMapping map[listenerKey]*listeners.Listener
+		isLBOwner          bool
+		lbName             string
+	}
+	tests := []struct {
+		name    string
+		fields  fields
```
Comment (dulek) [inline pkg/openstack/loadbalancer_test.go:550]: And this.

File: pkg/openstack/loadbalancer_test.go [hunk 3]
```diff
@@ -534,3 +534,158 @@ func Test_getListenerProtocol(t *testing.T) {
...
+					},
+				},
+				isLBOwner: true,
+				lbName:    "test-lb",
+			},
+			wantErr: false,
+			fields: fields{
+				LoadBalancer: LoadBalancer{},
+			},
+		},
```
Comment (dulek) [inline pkg/openstack/loadbalancer_test.go:645]: We should have a case when `lbOwner=true` and there are some tags on the Service.

**Review by dulek — CHANGES_REQUESTED** [review]

File: pkg/openstack/loadbalancer_test.go (hunk 1 above)
Comment (sakshi-1505) [reply pkg/openstack/loadbalancer_test.go:541]: It was added to mimic the interface of LoadBalancer, although we didn't use any field for the caller but it is actually needed to invoke the function from LoadBalancer interface.

Comment (dulek) [reply pkg/openstack/loadbalancer_test.go:541]: But no test case defines `LoadBalancer` as anything else than `LoadBalancer{}`. Can't we just define it in the loop that goes over the test cases?

**Review by dulek — CHANGES_REQUESTED** [review]

File: pkg/openstack/loadbalancer_test.go (hunk 1 above)
Comment (sakshi-1505) [reply pkg/openstack/loadbalancer_test.go:541]: Yes that makes sense, I will just define it in the loop

File: pkg/openstack/loadbalancer_test.go [hunk 4]
```diff
@@ -534,3 +534,158 @@ func Test_getListenerProtocol(t *testing.T) {
...
+						},
+					},
+				},
+				curListenerMapping: map[listenerKey]*listeners.Listener{
+					{
+						Protocol: "https",
+						Port:     9090,
+					}: {
+						ID:   "listenerid",
+						Tags: []string{"test-lb", "test-lb2"},
```
[the remark this answers is not in the document]
Comment (sakshi-1505) [reply pkg/openstack/loadbalancer_test.go]: fixed

File: pkg/openstack/loadbalancer_test.go (hunk 1 above)
Comment (sakshi-1505) [reply pkg/openstack/loadbalancer_test.go]: fixed

File: pkg/openstack/loadbalancer_test.go (hunk 2 above)
Comment (sakshi-1505) [reply pkg/openstack/loadbalancer_test.go]: fixed

File: pkg/openstack/loadbalancer_test.go (hunk 3 above)
Comment (sakshi-1505) [reply pkg/openstack/loadbalancer_test.go]: done

**Comment by sakshi-1505** [comment]
@pierreprinetti Please check this I have made all the changes which @dulek had requested

**Comment by pierreprinetti** [comment]
Thank you! Please rebase and we're good :rocket:

**Comment by sakshi-1505** [comment]
> Thank you! Please rebase and we're good 🚀

I think it's already aligned with `master` branch, can you please check again?

**Comment by pierreprinetti** [comment]
Right: I should have said “squash”, not “rebase”

**Comment by sakshi-1505** [comment]
There are multiple merge commits from main in this branch, squashing would
make the git history dirty due to those. Do you know how can I get over it?

**Comment by pierreprinetti** [comment]
The problem here is that you have **meged** master into your branch, whereas you should have **rebased** your branch onto master.

You have several options, the cleanest of which would entail `git fetch --all && git rebase -i origin/master`. I encourage you to explore that.

If you find yourself stuck, be aware that you can cancel the rebase at any time with `git rebase --abort`.

---

If instead you just want the job done and you're not afraid of using any means necessary:

```shell
git reset origin/master --hard \
	&& git apply <(curl -sSL https://github.com/kubernetes/cloud-provider-openstack/pull/2405.diff) \
	&& git add --all \
	&& git commit
```

But please don't tell anyone :shushing_face:

---

In any case, for the next time, I strongly advise you rebase instead of merging:

```shell
git fetch origin
git rebase origin/master
```

**Comment by pierreprinetti** [comment]
Unfortunately, another PR merged and you've won a new rebase round

**Comment by pierreprinetti** [comment]
Or maybe I am misinterpreting the label?
/lgtm
/assign jichenjc

**Comment by jichenjc** [comment]
/approve
/hold

hold for conflict solve, then let's unhold

**Comment by sakshi-1505** [comment]
@jichenjc @pierreprinetti  can you please re-add the lgtm labels & merge this?

**Pull request merged**